threshold_encryption/TEPrivateKeyShare.cpp

```diff
@@ -47,46 +41,31 @@ TEPrivateKeyShare::TEPrivateKeyShare( std::shared_ptr< std::string > _key_str, s
 
 TEPrivateKeyShare::TEPrivateKeyShare( libff::alt_bn128_Fr _skey_share, size_t _signerIndex,
     size_t _requiredSigners, size_t _totalSigners )
-    : privateKey( _skey_share ),
-      signerIndex( _signerIndex ),
-      requiredSigners( _requiredSigners ),
-      totalSigners( _totalSigners ) {
-    libBLS::ThresholdUtils::checkSigners( _requiredSigners, _totalSigners );
-
+    : TEBase( _requiredSigners, _totalSigners ),
+      privateKey( _skey_share ),
+      signerIndex( _signerIndex ) {
     if ( _signerIndex > _totalSigners ) {
         throw libBLS::ThresholdUtils::IncorrectInput( "Wrong _signerIndex" );
     }
 
-    libff::init_alt_bn128_params();
-
     if ( _skey_share.is_zero() ) {
         throw libBLS::ThresholdUtils::ZeroSecretKey( "Zero private key share" );
     }
 }
 
-libff::alt_bn128_G2 TEPrivateKeyShare::getDecryptionShare( libBLS::Ciphertext& cipher ) {
-    libBLS::TE::checkCypher( cipher );
-
-    libBLS::TE te( requiredSigners, totalSigners );
-
-    libff::alt_bn128_G2 decryption_share = te.getDecryptionShare( cipher, privateKey );
-
-    if ( decryption_share.is_zero() || !decryption_share.is_well_formed() ) {
-        throw libBLS::ThresholdUtils::IsNotWellFormed( "zero decrypt" );
-    }
-
-    return decryption_share;
+std::string TEPrivateKeyShare::toString() const {
+    return libBLS::ThresholdUtils::fieldElementToString( privateKey, 10 );
```

Comment:
What is 10 ?)) Please add a constexpr int that equals to 10, explaining what 10 means